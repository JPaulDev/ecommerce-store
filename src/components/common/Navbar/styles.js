import styled from 'styled-components';

export const Nav = styled.nav`
  background-color: var(--color-grey-4);
  box-shadow: 0 3px 2px 0 rgb(0 0 0 / 30%);
  position: relative;
  z-index: 90;
`;

export const NavList = styled.ul`
  display: flex;
  justify-content: center;
  list-style: none;
`;
