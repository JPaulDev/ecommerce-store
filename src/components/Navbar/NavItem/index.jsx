import { useState, useRef } from 'react';
import useOnClickOutside from '../../../hooks/useOnClickOutside';
import Dropdown from './Dropdown';
import * as Styled from './styles';

export default function NavItem({ item }) {
  const [isOpen, setIsOpen] = useState(false);
  const ref = useRef(null);

  const handleClose = () => setIsOpen(false);
  useOnClickOutside(ref, handleClose);

  return (
    <li ref={ref}>
      {item.hasDropdown ? (
        <Styled.Link
          as="button"
          type="button"
          isOpen={isOpen}
          onClick={() => setIsOpen(!isOpen)}
        >
          {item.title}
        </Styled.Link>
      ) : (
        <Styled.Link href={item.url}>{item.title}</Styled.Link>
      )}
      {isOpen && <Dropdown menu={item.title} />}
    </li>
  );
}
