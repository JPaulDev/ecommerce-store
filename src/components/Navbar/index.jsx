import NavItem from './NavItem';
import * as Styled from './styles';

const navItems = [
  {
    title: 'Home',
    url: '',
  },
  {
    title: 'Components',
    hasDropdown: true,
  },
  {
    title: 'Accessories',
    hasDropdown: true,
  },
  {
    title: 'About',
    url: '',
  },
  {
    title: 'Contact',
    url: '',
  },
];

export default function Navbar() {
  return (
    <Styled.Nav>
      <Styled.NavList>
        {navItems.map((item) => (
          <NavItem key={item.title} item={item} />
        ))}
      </Styled.NavList>
    </Styled.Nav>
  );
}
