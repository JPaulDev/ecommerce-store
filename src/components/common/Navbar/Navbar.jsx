import NavItem from './NavItem';
import * as Styled from './styles';

const navItems = [
  {
    heading: 'Home',
    href: '/',
  },
  {
    heading: 'Components',
    hasDropdown: true,
  },
  {
    heading: 'Accessories',
    hasDropdown: true,
  },
  {
    heading: 'About',
    href: '/about',
  },
  {
    heading: 'Contact',
    href: '/contact',
  },
];

export default function Navbar() {
  return (
    <Styled.Nav>
      <Styled.NavList>
        {navItems.map((item) => (
          <NavItem key={item.heading} item={item} />
        ))}
      </Styled.NavList>
    </Styled.Nav>
  );
}
