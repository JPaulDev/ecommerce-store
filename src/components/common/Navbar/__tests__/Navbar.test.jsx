import renderer from 'react-test-renderer';
import Navbar from '../index';

describe('Navbar component', () => {
  it('renders all links and buttons', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
