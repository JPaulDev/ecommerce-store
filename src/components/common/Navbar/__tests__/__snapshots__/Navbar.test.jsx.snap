// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders all links and buttons 1`] = `
<nav
  className="sc-32045fd3-0 glgyrw"
>
  <ul
    className="sc-32045fd3-1 cAfKXL"
  >
    <li>
      <a
        className="sc-571537e9-0 jvVTyW"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <button
        aria-controls="Components"
        aria-expanded={false}
        className="sc-571537e9-0 jvVTyW"
        onClick={[Function]}
        type="button"
      >
        Components
      </button>
    </li>
    <li>
      <button
        aria-controls="Accessories"
        aria-expanded={false}
        className="sc-571537e9-0 jvVTyW"
        onClick={[Function]}
        type="button"
      >
        Accessories
      </button>
    </li>
    <li>
      <a
        className="sc-571537e9-0 jvVTyW"
        href="/about"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        About
      </a>
    </li>
    <li>
      <a
        className="sc-571537e9-0 jvVTyW"
        href="/contact"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Contact
      </a>
    </li>
  </ul>
</nav>
`;
